src/providers/createProviderComponent.ts

```diff
@@ -0,0 +1,12 @@
+import { defineComponent } from 'vue';
+
+export function createProviderComponent(provideCb: () => any) {
+  return defineComponent({
+    setup() {
+      provideCb();
```

Comment:
I don't intuitively know what the 'Cb' stands for in this function name. I'd just rename it `provider`?
```suggestion
      provider();
```